projects/packages/my-jetpack/_inc/data/products/use-activate.ts

```diff
@@ -0,0 +1,35 @@
+import { __, sprintf } from '@wordpress/i18n';
+import { REST_API_SITE_PRODUCTS_ENDPOINT } from '../constants';
+import useSimpleMutation from '../use-simple-mutation';
+import useProduct from './use-product';
+
+const useActivate = ( productId: string ) => {
+	const { detail, refetch } = useProduct( productId );
+
+	const { mutate: activate, isPending } = useSimpleMutation(
+		'activateProduct',
+		{
+			path: `${ REST_API_SITE_PRODUCTS_ENDPOINT }/${ productId }`,
+			method: 'POST',
+		},
+		{
+			onSuccess: async () => {
+				// Await the refetch so the loading state is pending until this returns
+				await refetch();
+			},
```

Comment:
```suggestion
			onSuccess: refetch
```

that would be enough if we don't want to do any additional operations in this callback

Reply:
Nice catch, I thought I remembered this not working but it does 😅